### PR TITLE
Replace "matplotlibrc" by "rcParams" in the docs where applicable.

### PR DESCRIPTION
**Pull request by anntzer**
In preparation for the possible switch to the new rcsystem.

This doesn't touch any place actually documenting the rcsystem, just
places where "the foo param in the matplotlibrc" can reasonably be
replaced by "the foo rcParam".

The `examples.download` and `examples.directory` rcParams removed from
doc/matplotlibrc have been removed in 6c5e961 (2012).  The
verbose.level rcParam is deprecated, update the docs accordingly.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/ticks_and_spines/tick_xlabel_top.py [hunk 1]
```diff
@@ -7,9 +7,10 @@
 (default False) and :rc:`xtick.labelbottom` (default True) and
 :rc:`xtick.bottom` (default True) to control where on the axes ticks and
 their labels appear.
-These properties can also be set in the ``.matplotlib/matplotlibrc``.
 
+These properties can also be set in the ``.matplotlib/matplotlibrc``.
```
Comment (timhoffm) [inline examples/ticks_and_spines/tick_xlabel_top.py]: either "in ..." or "in the file ..."

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -3369,8 +3369,7 @@ def boxplot(self, x, notch=None, sym=None, vert=None, whis=None,
 
         """
 
-        # If defined in matplotlibrc, apply the value from rc file
-        # Overridden if argument is passed
+        # Missing arguments default to rcParam values.
```
Comment (timhoffm) [inline lib/matplotlib/axes/_axes.py]: rcParams

Comment (anntzer) [reply lib/matplotlib/axes/_axes.py]: fixed

File: examples/ticks_and_spines/tick_xlabel_top.py (hunk 1 above)
Comment (anntzer) [reply examples/ticks_and_spines/tick_xlabel_top.py]: fixed

**Pull request merged**